### PR TITLE
LUN-195: Update packages

### PR DESCRIPTION
**Pull request by Taggotty**
This pull request belongs to an issue on our [bugtracker](https://issues.tuerantuer.org/secure/RapidBoard.jspa?rapidView=27&view=detail).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword _LUN_.

jest update is blocked by jest-environment-enzyme and jest-enzyme and tbh. not urgent
As react-navigation and typescript were not much work for Lunes I integreated the update in this branch

### CODE REVIEW

File: src/hooks/useLoadFromEndpoint.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ export const loadFromEndpoint = async <T, P>(
     const response = await request(params)
     setData(response)
     setError(null)
-  } catch (e) {
+  } catch (e: any) {
```
Comment (Taggotty) [inline src/hooks/useLoadFromEndpoint.ts:24]: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-4.html#defaulting-to-the-unknown-type-in-catch-variables---useunknownincatchvariables

File: src/routes/DisciplineSelectionScreen.tsx [hunk 2]
```diff
@@ -23,9 +23,9 @@ const ItemText = styled.View`
   align-items: center;
 `
 
-const StyledList = styled(FlatList as new () => FlatList<DisciplineType>)`
+const StyledList = styled(FlatList)`
   width: 100%;
-`
+` as ComponentType as new () => FlatList<DisciplineType>
```
Comment (Taggotty) [inline src/routes/DisciplineSelectionScreen.tsx:28]: previous solution didn't work anymore

File: src/testing/createNavigationPropMock.ts [hunk 3]
```diff
@@ -1,16 +1,20 @@
-import { StackNavigationProp } from '@react-navigation/stack'
```
Comment (Taggotty) [inline src/testing/createNavigationPropMock.ts:1]: Mostly copied from integreat

**Review by f1sh1918 — APPROVED** [review]
Tested on android emulator and ios native. works without any issues! nice work

**Review by SteffiStuffel — APPROVED** [review]

File: src/routes/DisciplineSelectionScreen.tsx (hunk 2 above)
Comment (klinzo) [reply src/routes/DisciplineSelectionScreen.tsx:28]: why do we need the `as Flatlist` anyway?

**Review by klinzo — APPROVED** [review]
Not tested

File: src/routes/DisciplineSelectionScreen.tsx (hunk 2 above)
Comment (Taggotty) [reply src/routes/DisciplineSelectionScreen.tsx:28]: To type the generic

**Pull request merged**